plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2711,7 +2715,7 @@ void producer_plugin_impl::switch_to_read_window() {
 bool producer_plugin_impl::read_only_trx_execution_task() {
    auto start = fc::time_point::now();
    auto read_window_deadline = start + _ro_read_window_effective_time_us;
-   while ( fc::time_point::now() < read_window_deadline ) {
+   while ( fc::time_point::now() < read_window_deadline && !_received_block ) {
```

Comment:
`_recevied_block` needs to be set to `false` before the start of all the `read_only_trx_execution_task`.

Reply:
`net_plugin` is setting it to `true` every time a block comes in. You are interested in if a block has come in since you started your read only window. Yes there is a race there between end of write window and start of read window, just like there is a race between the start_block and a block coming in.

Reply:
Thanks for the explannation.

Reply:
Done. Thanks.